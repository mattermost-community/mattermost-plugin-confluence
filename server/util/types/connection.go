package types

type User struct {
	MattermostUserID string `json:"mattermost_user_id"`
	InstanceURL      string `json:"instance_url,omitempty"`
}

type ConfluenceUser struct {
	AccountID   string `json:"accountId,omitempty"`
	Name        string `json:"username,omitempty"`
	DisplayName string `json:"displayName,omitempty"`
}

type UserGroups struct {
	Groups []*UserGroup `json:"results,omitempty"`
}

type UserGroup struct {
	Name string `json:"name"`
}

type Connection struct {
	ConfluenceUser
	OAuth2Token       string `json:"token,omitempty"`
	DefaultProjectKey string `json:"default_project_key,omitempty"`
	IsAdmin           bool   `json:"is_admin,omitempty"`
	MattermostUserID  string `json:"mattermost_user_id,omitempty"`
}

func (c *Connection) ConfluenceAccountID() string {
	if c.AccountID != "" {
		return c.AccountID
	}

	return c.Name
}

func NewUser(mattermostUserID string) *User {
	return &User{
		MattermostUserID: mattermostUserID,
	}
}

func (user *User) AsConfigMap() map[string]interface{} {
	return map[string]interface{}{
		"mattermost_user_id": user.MattermostUserID,
		"instance_url":       user.InstanceURL,
	}
}
