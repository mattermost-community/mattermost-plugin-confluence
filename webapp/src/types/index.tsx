export type ConfluenceConfig = {
    serverURL: string,
    clientID: string,
    clientSecret: string
}

export type ReactSelectOption = {
    label: string | React.ReactElement;
    value: string;
};

export type ErrorResponse = {
    response: {
        text: string;
    }
};
