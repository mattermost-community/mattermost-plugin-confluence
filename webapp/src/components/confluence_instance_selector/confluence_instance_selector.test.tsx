import React from 'react';

import {shallow} from 'enzyme';

import {Provider} from 'react-redux';

import {Theme} from 'mattermost-redux/types/preferences';

import {configureStore} from 'tests/setup';

import ConfluenceInstanceSelector from '.';

describe('components/ConfluenceInstanceSelector', () => {
    const initialState = {};
    const baseProps = {
        theme: {} as Theme,
        selectedInstanceID: 'test-spaceKey',
        onInstanceChange: jest.fn(),
    };
    const mockStore = configureStore();
    test('confluence instance selector snapshot test', async () => {
        const props = {
            ...baseProps,
        };
        const store = mockStore(initialState);
        const wrapper = shallow(
            <Provider store={store}>
                <ConfluenceInstanceSelector {...props}/>
            </Provider>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
