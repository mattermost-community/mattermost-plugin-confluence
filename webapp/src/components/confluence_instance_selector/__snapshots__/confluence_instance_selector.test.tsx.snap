// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ConfluenceInstanceSelector confluence instance selector snapshot test 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ConfluenceInstanceSelector
    onInstanceChange={[MockFunction]}
    selectedInstanceID="test-spaceKey"
    theme={Object {}}
  />
</ContextProvider>
`;
